static_analyzer/__init__.py

```
@@ -636,103 +636,161 @@ def _update_cached_results(
-            # Scope the repo-wide set to this language's project root so a
-            # multi-language repo doesn't re-LSP every changed file per engine.
-            return {f for f in self.changed_files if f.is_relative_to(project_path)}
+            return {path.resolve() for path in self.changed_files}
         try:
             return set(get_changed_files_since(project_path, cached_sha))
-        except Exception as e:
-            logger.warning(
-                f"get_changed_files_since failed for {language} (cached_sha={cached_sha}): {e}; "
-                "falling back to full re-LSP for this language"
-            )
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)